contracts/contracts/strategies/Generalized4626Strategy.sol

```diff
@@ -112,6 +119,7 @@ contract Generalized4626Strategy is InitializableAbstractStrategy {
     function checkBalance(address _asset)
         external
         view
+        virtual
```

Comment:
I can't comment on line 133 so I'll comment here. `IERC4626(platformAddress).convertToAssets` needs to be replaced with `IERC4626(platformAddress).maxWithdraw`.
`convertToAssets` does not include fees or slippage. It's just an estimate for display purposes.

Reply:
For @naddison36, @shahthepro, @sparrowDom, and all who have wondered about the mystery why we seem to be getting slightly different numbers back from `maxWithdraw` and `convertToAssets`, even though the underlying code path appears the same, I think I've solved the issue.

Basically, `maxWithdraw` and `convertToAssets` do actually return the same thing - differences seen are because the number changes block by block, and manually finding and clicking things on etherscan will tend to display numbers pulled from different blocks.

Here's the result of a fork simulation in forge:

```python
  Reported: maxWithdraw     2014259136521724599573
  Reported: convertToAssets 2014259136521724599573
  Actual:   redeemed        2014259136521724599573
  ```

Test code:

```solidity
contract WithdrawTest is Test {
    
    IsfrxETH sfrxETH = IsfrxETH(0xac3E018457B222d93114458476f3E3416Abbe38F);
    address strat = 0x3fF8654D633D4Ea0faE24c52Aec73B4A20D0d0e5;

    function testReportedAmounts() external {
        uint256 shareBalance = sfrxETH.balanceOf(strat);
        console2.log("Share balance", shareBalance);
        console.log("--------");
        console2.log("Reported: maxWithdraw    ", sfrxETH.maxWithdraw(strat));
        console2.log("Reported: convertToAssets", sfrxETH.convertToAssets(shareBalance));
        vm.prank(strat);
        uint256 x = sfrxETH.redeem(shareBalance, strat, strat);
        console2.log("Actual:   redeemed       ", x);
    }
}
```

Reply:
For the sfrxETH vault, `maxWithdraw` returns the same as `convertToAssets`. But `Generalized4626Strategy` should work with all ERC-4626 vaults which means it should work with vaults where `convertToAssets` does not match `maxWithdraw`. That is, it should work with vaults that either have slippage or fees.

Reply:
I'm fine with maxWithdraw on this one.